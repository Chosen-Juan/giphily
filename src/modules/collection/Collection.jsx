import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';

import CollectionDropdown from '../../modules/collectionDropdown/CollectionDropdown';
import Gif from '../gifList/Gif';
import { removeGifFromCollection } from './CollectionActions';
import './styles.scss';

export class Collection extends Component {

  static propTypes = {
    removeGifFromCollectionDispatcher: PropTypes.func.isRequired,
    collection: PropTypes.arrayOf(Gif.WrappedComponent.propTypes.gif)
  };

  static defaultProps = {
    collection: []
  };

  renderGifs(collection) {
    return collection.map(gif => {
      const removeFromCollection = () => {
        this.props.removeGifFromCollectionDispatcher(gif.id);
      };
      return (
        <div key={gif.id} className='gif-wrap'>
          <div className='gif-button' onClick={removeFromCollection}>
            <span>Remove</span>
          </div>
          <Gif gif={gif} />
        </div>
      );
    });
  }

  render() {
    return (
      <section className='collection-container'>
        <CollectionDropdown />
        <div className='collection'>
            {this.renderGifs(this.props.collection)}
        </div>
      </section>
    );
  }
};

const mapStateToProps = ({ collectionReducer: { items } }) => ({
  collection: items
});

const mapDispatchToProps = dispatch => ({
  removeGifFromCollectionDispatcher: index => dispatch(removeGifFromCollection(index))
});

export default connect(mapStateToProps, mapDispatchToProps)(Collection);
