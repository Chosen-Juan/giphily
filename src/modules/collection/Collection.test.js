import React from 'react';
import { shallow } from 'enzyme';

import { Collection } from './Collection';

const removeGifSpy = jest.fn();
const defaultProps = {
  removeGifFromCollectionDispatcher: removeGifSpy,
};


describe('Collection', () => {
  it('should match snapshot', () => {
    const collection = shallow(<Collection {...defaultProps} />);
    expect(collection).toMatchSnapshot();
  });
});
