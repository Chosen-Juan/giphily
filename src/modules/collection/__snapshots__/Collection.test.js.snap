// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection should match snapshot 1`] = `
<section
  className="collection-container"
>
  <CollectionDropdown />
  <div
    className="collection"
  />
</section>
`;
